compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaLexer.java

```diff
@@ -1160,11 +1160,25 @@ private STToken processExclamationMarkOperator() {
                     return getSyntaxToken(SyntaxKind.NOT_EQUAL_TOKEN);
                 }
             default:
+                // this is '!is'
+                if (isNotIsToken()) {
+                     return getSyntaxToken(SyntaxKind.NOT_IS_KEYWORD);
+                 }
                 // this is '!'
                 return getSyntaxToken(SyntaxKind.EXCLAMATION_MARK_TOKEN);
         }
     }
 
+    private boolean isNotIsToken() {
+        if ((reader.peek() == 'i' && reader.peek(1) == 's') &&
+                !(isIdentifierFollowingChar(reader.peek(2)) || reader.peek(2) == LexerTerminals.BACKSLASH)) {
+            reader.advance(2);
```

Comment:
Shall we move `reader.advance(2);` to L1165?  As the method name implies it may just be about checking a condition.

Reply:
Let's add a parser test for `is` followed by a letter, backslash cases.  
e.g. `!isNotFound`, `!is\ not\ found`
```
if !is\ not\ found {
            
}
```

Reply:
Negative syntax parser test for space between `!` and `is` can also be added.
```
if a ! is int {
                
}
```

Reply:
okay. Will add